### PR TITLE
Allow resetting FPS listener

### PR DESCRIPTION
**Pull request by tobrun**
### Summary of changes
This PR allows to reset the OnFpsChangedListener. This allows users to stop listening to these types of event and avoid leaking any objects. This change also performs automatic cleanup as part of onDestroy. 

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

### CODE REVIEW

**Review by jush — APPROVED** [review]

File: sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderer.kt [hunk 1]
```diff
@@ -47,6 +47,7 @@ internal abstract class MapboxRenderer : MapClient {
   @UiThread
   fun onDestroy() {
     logI(TAG, "onDestroy")
+    setOnFpsChangedListener(null)
```
[the remark this answers is not in the document]
Comment (kiryldz) [reply sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderer.kt:50]: the same code happens on line 52 - thus this is not needed

[the remark this answers is not in the document]
Comment (alexander-kulikovskii) [reply sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderer.kt:50]: maybe not only this case.
when user set some listener, then set it to `null` and check that everything works and you don't receive any data.

File: sdk/api/metalava.txt [hunk 2]
```diff
@@ -12,7 +12,7 @@ package com.mapbox.maps {
     method public void removeRendererSetupErrorListener(com.mapbox.maps.renderer.RendererSetupErrorListener rendererSetupErrorListener);
     method @com.mapbox.maps.MapboxExperimental public boolean removeWidget(com.mapbox.maps.renderer.widget.Widget widget);
     method public void setMaximumFps(int fps);
-    method public void setOnFpsChangedListener(com.mapbox.maps.renderer.OnFpsChangedListener listener);
```
Comment (pengdev) [inline sdk/api/metalava.txt:15]: interesting that the binary compatibility validator didn't seem to care about the nullability changes, according to https://github.com/Kotlin/binary-compatibility-validator/issues/20

however, shall we consider the risk of breaking changes here?

**Review by kiryldz — COMMENTED** [review]
would be great to add the unit test setting NULL
